kyaml/kio/filters/testyaml/testyaml.go

```diff
@@ -36,6 +36,20 @@ var UnformattedJSON1 = []byte(`
 }
 `)
 
+var UnformattedYamlWithNoFmtAnnotation = []byte(`
+spec: a
+status:
+  conditions:
+  - 3
+  - 1
+  - 2
+apiVersion: example.com/v1beta1
+kind: MyType
+metadata:
+  annotations:
+    config.kubernetes.io/no-fmt: true
```

Comment:
How about something less boolean:

```
config.kubernetes.io/formatting: none
```
with default value `standard`, allowing future options like
 `commentsStripped`, `tabsinsteadofspaces`, `lexSort`, etc.

Then: 
- no anno means do `standard` formatting.
- anno present but with something other than `none` or `standard` is an error

Reply:
This is a good suggestion. I have updated the PR.